apps/api/src/app/events/usecases/digest-filter-steps/digest-filter-steps.usecase.ts

```diff
@@ -46,4 +46,25 @@ export class DigestFilterSteps {
       _templateId: command.templateId,
     };
   }
+
+  public static getNestedValue<ObjectType>(payload: ObjectType, path?: string) {
+    if (!path || !payload) {
+      return undefined;
+    }
+
+    try {
+      let result;
+      const keys = path.split('.');
+
+      for (const key of keys) {
+        result = payload[key];
+      }
+
+      return result;
+    } catch (error) {
+      Logger.error('Failure when parsing digest payload nested key');
+
+      return undefined;
+    }
```

Comment:
I think we can use `lodash.get` using a similar approach introduced in: https://github.com/novuhq/novu/pull/2794 it will also support arrays and etc...